Android/unityLibrary/hybidwrapper.androidlib/src/main/java/net/pubnative/unityplugin/HyBidAdViewWrapper.java

```diff
@@ -0,0 +1,130 @@
+
+public class HyBidAdViewWrapper implements HyBidAdView.Listener {
+
+    private static final String TAG = HyBidAdViewWrapper.class.getSimpleName();
+
+    protected HyBidAdView hyBidAdView;
+
+    protected String mGameObjectName;
+    protected String mAdId;
+
+    protected void executeDisplayAction(Runnable runnable) {
+        if (UnityPlayer.currentActivity == null) {
+            Log.e(TAG, "No active context found to show the ad");
+        } else {
+            UnityPlayer.currentActivity.runOnUiThread(runnable);
+        }
+    }
+
+    protected void setGameObject(String gameObjectName) {
+        this.mGameObjectName = gameObjectName;
+    }
+
+    protected void setAdId(String adId) {
+        this.mAdId = adId;
+    }
+
+    public void load(String gameObjectName, String appToken, final String placementId, String adId, int position) {
+        if (ContextCompat.checkSelfPermission(UnityPlayer.currentActivity.getApplicationContext(),
+                Manifest.permission.ACCESS_FINE_LOCATION) == PackageManager.PERMISSION_GRANTED) {
+            doHyBidStuff(gameObjectName, appToken, placementId, adId, position);
+        } else {
+            ActivityCompat.requestPermissions(UnityPlayer.currentActivity,
+                    new String[]{Manifest.permission.ACCESS_FINE_LOCATION},
+                    10);
+        }
+    }
+
+    private void doHyBidStuff(String gameObjectName, final String appToken, final String placementId, String adId, final int position) {
+
+        try {
+            setGameObject(gameObjectName);
+            setAdId(adId);
+
+            if (hyBidAdView != null) {
+                UnityPlayer.currentActivity.runOnUiThread(new Runnable() {
+                    @Override
+                    public void run() {
+                        hyBidAdView.destroy();
+                        loadBanner(appToken, placementId, position);
+                    }
+                });
+            } else {
+                loadBanner(appToken, placementId, position);
+            }
+        } catch (Exception e) {
+            Log.d("Exception", e.toString());
+        }
+    }
+
+    private void loadBanner(String appToken, final String placementId, int position) {
+        final HyBidAdView.Position bannerPosition;
+
+        if (position == getTopPosition()) {
+            bannerPosition = HyBidAdView.Position.TOP;
+        } else {
+            bannerPosition = HyBidAdView.Position.BOTTOM;
+        }
+
+        if (UnityPlayer.currentActivity == null) {
+            Log.e(TAG, "No active context found to load the interstitial");
+        } else {
+            if (!HyBid.isInitialized()) {
+                HyBid.initialize(appToken, UnityPlayer.currentActivity.getApplication(), new HyBid.InitialisationListener() {
+                    @Override
+                    public void onInitialisationFinished(boolean b) {
+                        hyBidAdView = new HyBidAdView(UnityPlayer.currentActivity);
```

Comment:
code duplication. You could move it into a separate func and to call it then

Reply:
There is only 1 method loadBanner and it is called in all cases 
I already did it in last commit

Reply:
I was talking about those lines:

```
hyBidAdView = new HyBidAdView(UnityPlayer.currentActivity);
hyBidAdView.setAdSize(AdSize.SIZE_320x50);
hyBidAdView.load(placementId, bannerPosition, HyBidAdViewWrapper.this);
```